doc/source/data/working-with-llms.rst

```diff
@@ -33,72 +33,34 @@ The :class:`vLLMEngineProcessorConfig <ray.data.llm.vLLMEngineProcessorConfig>`
 It contains the model name, the number of GPUs to use, and the number of shards to use, along with other vLLM engine configurations.
 Upon execution, the Processor object instantiates replicas of the vLLM engine (using :meth:`map_batches <ray.data.Dataset.map_batches>` under the hood).
 
-.. testcode::
-
-    import ray
-    from ray.data.llm import vLLMEngineProcessorConfig, build_llm_processor
-    
-    config = vLLMEngineProcessorConfig(
-        model_source="unsloth/Llama-3.1-8B-Instruct",
-        engine_kwargs={
-            "enable_chunked_prefill": True,
-            "max_num_batched_tokens": 4096,
-            "max_model_len": 16384,
-        },
-        concurrency=1,
-        batch_size=64,
-    )
-    processor = build_llm_processor(
-        config,
-        preprocess=lambda row: dict(
-            messages=[
-                {"role": "system", "content": "You are a bot that responds with haikus."},
-                {"role": "user", "content": row["item"]}
-            ],
-            sampling_params=dict(
-                temperature=0.3,
-                max_tokens=250,
-            )
-        ),
-        postprocess=lambda row: dict(
-            answer=row["generated_text"],
-            **row  # This will return all the original columns in the dataset.
-        ),
-    )
-
-    ds = ray.data.from_items(["Start of the haiku is: Complete this for me..."])
-
-    ds = processor(ds)
-    ds.show(limit=1)
-
-.. testoutput::
-    :options: +MOCK
-
-    {'answer': 'Snowflakes gently fall\nBlanketing the winter scene\nFrozen peaceful hush'}
+.. literalinclude:: doc_code/working-with-llms/basic_llm_example.py
+    :language: python
+    :start-after: __basic_llm_example_start__
+    :end-before: __basic_llm_example_end__
+
+Here's a simple configuration example:
+
+.. literalinclude:: doc_code/working-with-llms/basic_llm_example.py
+    :language: python
+    :lines: 28-37
 
 Each processor requires specific input columns. You can find more info by using the following API:
 
-.. testcode::
+The processor requires specific input columns. Here's how to check what columns are needed:
 
-    processor.log_input_column_names()
+.. literalinclude:: doc_code/working-with-llms/basic_llm_example.py
+    :language: python
+    :lines: 65-67
```

Comment:
Let's not use lines directive? and use the tags. So add as many ___xxx___ tags as you need to the source code so you can ref them here.